pkg/antctl/raw/multicluster/join/join.go

```diff
@@ -0,0 +1,368 @@
+  clusterset: test-clusterset
+leaderCluster:
+  namespace: antrea-multicluster
+  secret: member-access-token
+  server: https://172.18.0.3:6443
+  clusterid: test-cluster-north
+  clusterset: test-clusterset
+`, "\n")
+
+func NewJoinCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := joinOptions{}
+	options = &o
+	command.Flags().StringVarP(&options.LeaderClusterOpt.Namespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&options.LeaderClusterOpt.ClusterID, "leader-cluster-id", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&options.LeaderClusterOpt.Secret, "secret", "", "", "Secret of the leader cluster")
+	command.Flags().StringVarP(&options.LeaderClusterOpt.Server, "leader-server", "", "", "Server to the leader cluster")
+	command.Flags().StringVarP(&options.MemberClusterOpt.ClusterSet, "cluster-set", "", "", "ClusterSet to join")
+	command.Flags().StringVarP(&options.MemberClusterOpt.Namespace, "namespace", "", "", "Namespace of the member cluster")
+	command.Flags().StringVarP(&options.MemberClusterOpt.ClusterID, "cluster-id", "", "", "Cluster ID of the member cluster")
+	command.Flags().StringVarP(&options.ConfigFile, "config", "f", "", "Config file of the member cluster")
+
+	return command
+}
+
+func joinRunE(cmd *cobra.Command, args []string) error {
+	if err := options.validateAndComplete(); err != nil {
+		return err
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restConfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restConfigTmpl)
+
+	k8sClient, err := client.New(restConfigTmpl, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	memberClusterNamespace := options.MemberClusterOpt.Namespace
+	memberClusterID := options.MemberClusterOpt.ClusterID
+	memberClusterSet := options.MemberClusterOpt.ClusterSet
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+	var createErr error
+	if createErr = k8sClient.Create(context.TODO(), newClusterClaim(memberClusterID, memberClusterNamespace)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimID, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+		defer func() {
```

Comment:
How could this defer works if it's only wrapped in the `else` case `createErr == nil `?